pkg/controller/clusterdeployment/clusterprovisions.go

```diff
@@ -553,34 +559,65 @@ func (r *ReconcileClusterDeployment) copyInstallLogSecret(destNamespace string,
 	return controllerutils.CopySecret(r, src, dest, nil, nil)
 }
 
-func getInstallLogEnvVars(secretPrefix string) []corev1.EnvVar {
-	extraEnvVars := []corev1.EnvVar{}
+// NOTE: Ugly-but-simple way to mock ioutil.ReadFile for test purposes.
+// https://stackoverflow.com/questions/20923938/how-would-i-mock-a-call-to-ioutil-readfile-in-go/37035375
+// This variable is overridden by fakeReadFile.
+var readFile = ioutil.ReadFile
 
-	cloudProvider, found := os.LookupEnv(constants.InstallLogsUploadProviderEnvVar)
-	if !found {
-		return extraEnvVars
+// readProvisionFailedConfig reads the provision fail config from the file pointed to
+// by the FailedProvisionConfigFileEnvVar environment variable.
+func readProvisionFailedConfig() (*hivev1.FailedProvisionConfig, error) {
+	path := os.Getenv(constants.FailedProvisionConfigFileEnvVar)
+	if len(path) == 0 {
+		return nil, nil
 	}
 
-	extraEnvVars = append(extraEnvVars, corev1.EnvVar{
-		Name:  constants.InstallLogsUploadProviderEnvVar,
-		Value: cloudProvider,
-	})
+	config := &hivev1.FailedProvisionConfig{}
+
+	fileBytes, err := readFile(path)
+	if err != nil || len(fileBytes) == 0 {
+		return config, err
+	}
+	if err := json.Unmarshal(fileBytes, config); err != nil {
+		return config, err
+	}
+
+	return config, nil
+}
 
-	if cloudProvider == constants.InstallLogsUploadProviderAWS {
-		secretName, foundSrc := os.LookupEnv(constants.InstallLogsCredentialsSecretRefEnvVar)
-		if foundSrc {
-			extraEnvVars = append(extraEnvVars, corev1.EnvVar{
+func getInstallLogEnvVars(secretPrefix string) ([]corev1.EnvVar, error) {
+	var extraEnvVars = []corev1.EnvVar{}
+	fpConfig, err := readProvisionFailedConfig()
+	if err != nil || fpConfig == nil {
+		return extraEnvVars, err
+	}
+	if awsSpec := fpConfig.AWS; awsSpec != nil {
+		// By default we will try to gather logs on failed installs:
+		extraEnvVars = []corev1.EnvVar{
+			{
+				Name:  constants.InstallLogsUploadProviderEnvVar,
+				Value: constants.InstallLogsUploadProviderAWS,
+			},
+			{
 				Name:  constants.InstallLogsCredentialsSecretRefEnvVar,
-				Value: secretPrefix + "-" + secretName,
-			})
+				Value: secretPrefix + "-" + awsSpec.CredentialsSecretRef.Name,
+			},
+			{
+				Name:  constants.InstallLogsAWSRegionEnvVar,
+				Value: awsSpec.Region,
+			},
+			{
+				Name:  constants.InstallLogsAWSServiceEndpointEnvVar,
+				Value: awsSpec.ServiceEndpoint,
+			},
+			{
+				Name:  constants.InstallLogsAWSS3BucketEnvVar,
+				Value: awsSpec.Bucket,
+			},
 		}
-
-		extraEnvVars = addEnvVarIfFound(constants.InstallLogsAWSRegionEnvVar, extraEnvVars)
```

Comment:
I couldn't find any more users of this `addEnvVarIfFound` function. can we just remove it as well?

Reply:
Good eye. Done.